lib/events.js

```diff
@@ -26,25 +29,57 @@ $(document).ready(function() {
     const $lunchRemainingCal = $('#lunch-remaining-calories')
     const $snackRemainingCal = $('#snack-remaining-calories')
     const $dinnerRemainingCal = $('#dinner-remaining-calories')
+    const $brfastGoalCal = $('#breakfast-goal-calories')
+    const $lunchGoalCal = $('#lunch-goal-calories')
+    const $snackGoalCal = $('#snack-goal-calories')
+    const $dinnerGoalCal = $('#dinner-goal-calories')
     const $totalCal = $('#total-calories')
     const $remainingCal = $('#remaining-calories')
     const $goalCal = $('#goal-calories')
+    const $mealButtons = $('.meal-button')
+    const $mealTables = $('.meal-tables')
 
+    const foodIncluded = (node, query) => {
+        return $(node)
+            .find('td')
+            .first()
+            .text()
+            .toLowerCase()
+            .includes(query)
+    }
 
-    Food.getFoods()
+    const sortFoodTable = (direction) => {
+        $('#diary-food-index tr').sort(function(a, b) {
+            switch (direction) {
+                case 'desc':
+                    return $('td:nth-child(2)', b).text().localeCompare($('td:nth-child(2)', a).text());
+                    break;
+                case 'orig':
+                    return Number(a.id) - Number(b.id)
+                    break;
+                case 'asc':
+                    return $('td:nth-child(2)', a).text().localeCompare($('td:nth-child(2)', b).text());
+                    break;
+            }
+        }).appendTo($('#diary-food-index'))
+    }
+
+
+    Food.getAll()
         .then(function(response) {
             return response.map(function(food) {
                 return new Food(food)
             })
         })
         .then(function(foods) {
             foods.forEach(function(food) {
-                $foodIndex.prepend(food.toHTML())
+                $foodIndex.append(food.toHTML())
+                $diaryFoodIndex.append(food.toDiaryHTML())
```

Comment:
good catch on the ordering and finding the relation to the default scope from the db.